app/containers/TuneContainer/saga.js

```diff
@@ -12,10 +15,37 @@ export function* getItuneSongs(action) {
   if (ok) {
     yield put(successGetItuneSongs(data));
   } else {
-    yield put(failureGetItuneSongs(data));
+    const error = data?.originalError?.message ?? translate('something_went_wrong');
+    yield put(failureGetItuneSongs(error));
   }
 }
 
+export function* getTrackDetails(action) {
```

Comment:
This seems a tad bit overdone. Do you think a simplification is possible? We need to check if the redux store has `songsData.results` and take the result from the id we passed. I think it can be simplified a bit.